examples/MQ/pixelDetector/src/PixelDigiBinSource.h

```diff
@@ -62,8 +59,8 @@ class PixelDigiBinSource : public FairSource
     Bool_t ActivateObject(TObject** obj, const char* BrName) override;
 
   private:
-    PixelEventHeader* fEventHeader;
-    TClonesArray* fDigis; /** Output array of PixelDigi **/
+    PixelEventHeader fEventHeader{};
+    TClonesArray fDigis; /** Output array of PixelDigi **/
```

Comment:
Are you sure that this will work in cases where the TClonesArray needs to be resized?

Reply:
https://root.cern/doc/master/classTClonesArray.html#a2bb70b2ac885b9120dcc304fde2cfd50

> The second argument, s, indicates an approximate number of objects that will be entered in the array. If more than s objects are entered, the array will be automatically expanded.

How does a `new`/`delete`d object change that?

Reply:
If ROOT can enlarge the TClonesArray in place there is probably no problem. Bu in general case ROOT needs to create anew and larger TClonesArray somewhere in memory, copy the old data, change the pointer to the new location and deletes the old TClonesArray.

Reply:
Hm, but why would it ever need to create a new TCA? It would need to point the [`TClonesArray::fCont` member](https://root.cern/doc/master/classTObjArray.html#a1f358720e4cf05c1031d48e20824729e) to the new storage location and perhaps update `TClonesArray::fLast/fLowerBound`, would it not?

Reply:
and second question: how would ROOT in general know what to do with the old TCAs? Does ROOT own all TCAs then?

Reply:
The TCA object is always 80 bytes in size (on x86-64), no matter how many elements are in there. The element storage is somewhere else. So it shouldn't make a difference, where the TCA lives.

Reply:
Right. I did not check the code properly. You are right, ROOT is handling the memory itself. So forget my comment.
It look like the TCA is implemented in the same way as e.g. the vector.